org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/contentassist/ConfigurableCompletionProposal.java

```diff
@@ -596,9 +597,14 @@ public Object getAdditionalProposalInfo(IProgressMonitor monitor) {
 				eObject = (EObject) additionalProposalInfo;
 			} else { 
 				if (additionalProposalInfo instanceof Provider) {
-					Object o = ((Provider<?>) additionalProposalInfo).get();
-					if (o instanceof EObject)
-						eObject = (EObject) o;
+					try {
+						Object o = ((Provider<?>) additionalProposalInfo).get();
+						if (o instanceof EObject)
+							eObject = (EObject) o;
+					} catch (OperationCanceledException e) {
+						monitor.setCanceled(true);
```

Comment:
Is this the common pattern?

Reply:
i have doubts this is the correct solution

Reply:
simply set o to null?

Reply:
I second Christians concerns here. Usually the caller of a method that accepts an IProgressMonitor also describes the contract for the cancellation. Especially it should be able to deal with OCE.